src/configuration_server.erl

```diff
@@ -68,13 +68,20 @@ get_config_file_path() ->
     {ok, State :: #configuration{}} |
     {ok, State :: #configuration{}, timeout() | hibernate} |
     {stop, Reason :: term()} | ignore.
-init([]) ->
-    ConfigFilePath = search_for_config_file_path(
+init([no_path]) ->
+    ConfigFilePathResult = search_for_config_file_path(
         ?TEP_DEFAULT_CONFIG_LOCATIONS,
         ?TEP_DEFAULT_CONFIG_FILE_NAMES
     ),
-    Configuration = create_configuration(ConfigFilePath),
-    return_from_init(Configuration).
+    ConfigurationResult = create_configuration(ConfigFilePathResult),
+    return_from_init(ConfigurationResult);
+init([ConfigFilePath]) ->
+    ConfigFilePathResult = search_for_config_file_path(
+        ["./"],
+        [ConfigFilePath]
+    ),
+    ConfigurationResult = create_configuration(ConfigFilePathResult),
+    return_from_init(ConfigurationResult).
```

Comment:
Hoo boy, my TypeScript eyes just wanna
```typescript
export function init(configFilePath: string | undefined) {
  const configFilePathResult = searchForConfigFilePath(
    configFilePath ? 
      { searchDir: './', fileName: configFilePath } :
      { searchDir: DEFAULT/* stuff */, ... }
  );
  // ...
}
```
soooo bad. I know what we have is probably "right" in the sense that it's Erlang idiomatic, but I'm still getting used to what that means, honestly...

Reply:
I don't know why we created this repeating mess, but I have implemented a simpler and cleaner version.